data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/codec/OTelProtoCodec.java

```diff
@@ -0,0 +1,270 @@
+import com.amazon.dataprepper.model.trace.Span;
+import com.amazon.dataprepper.model.trace.SpanEvent;
+import com.amazon.dataprepper.model.trace.TraceGroupFields;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.InstrumentationLibrary;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Status;
+import org.apache.commons.codec.binary.Hex;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * OTelProtoCodec parses the {@link ExportTraceServiceRequest} into Collection<{@link Span}>.
+ * <p>
+ */
+public class OTelProtoCodec {
+    private static final ObjectMapper OBJECT_MAPPER =  new ObjectMapper();
+    private static final String SERVICE_NAME = "service.name";
+    private static final String SPAN_ATTRIBUTES = "span.attributes";
+    static final String RESOURCE_ATTRIBUTES = "resource.attributes";
+    static final String INSTRUMENTATION_LIBRARY_NAME = "instrumentationLibrary.name";
+    static final String INSTRUMENTATION_LIBRARY_VERSION = "instrumentationLibrary.version";
+    static final String STATUS_CODE = "status.code";
+    static final String STATUS_MESSAGE = "status.message";
+    /**
+     * To make it ES friendly we will replace '.' in keys with '@' in all the Keys in {@link io.opentelemetry.proto.common.v1.KeyValue}
+     */
+    private static final String DOT = ".";
+    private static final String AT = "@";
+    public static final Function<String, String> REPLACE_DOT_WITH_AT = i -> i.replace(DOT, AT);
+    /**
+     * Span and Resource attributes are essential for OpenSearch so they should not be nested. SO we will prefix them with "span.attributes"
+     * and "resource.attributes".
+     *
+     */
+    public static final Function<String, String> SPAN_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> SPAN_ATTRIBUTES + DOT + i.replace(DOT, AT);
+    public static final Function<String, String> RESOURCE_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> RESOURCE_ATTRIBUTES + DOT + i.replace(DOT, AT);
+
+    public static List<Span> parseExportTraceServiceRequest(final ExportTraceServiceRequest exportTraceServiceRequest) {
+        return exportTraceServiceRequest.getResourceSpansList().stream()
+                .flatMap(rs -> parseResourceSpans(rs).stream()).collect(Collectors.toList());
+    }
+
+    public static List<Span> parseResourceSpans(final ResourceSpans resourceSpans) {
```

Comment:
Regarding my comment on non-static, I'm fine with keeping any private methods static.

Reply:
I will make it protected. 

Just a note: OTelProtoCodec will be further refactored into common plugins in the following PR as for the convenience of maintenance on consistency between encoding (peer-forwarder client) and decoding (otel-trace-source). So this will not be the final version of it.